src/main/java/com/targomo/client/api/response/DefaultResponse.java

```diff
@@ -22,26 +23,39 @@
     private long roundTripTimeMillis;
     private TravelOptions travelOptions;
 
-    public void setExtraParameters(TravelOptions travelOptions, long roundTripTimeMillis, long parseTimeMillis) {
+    public static <R extends DefaultResponse> R createGatewayTimeoutResponse(Class<R> clazz) throws TargomoClientException {
+        try {
+            R gateWayTimeoutResponse = clazz.newInstance();
+            gateWayTimeoutResponse.setCode("gateway-time-out");
+            gateWayTimeoutResponse.setMessage("");
+            gateWayTimeoutResponse.setRequestTimeMillis(-1);
+            return gateWayTimeoutResponse;
+        } catch (InstantiationException | IllegalAccessException e) {
+            throw new TargomoClientException("Response Instantiation failed with error", e);
+        }
+    }
+
+    public void finishDeserialization(TravelOptions travelOptions, long roundTripTimeMillis, long parseTimeMillis) {
         this.travelOptions 	   	 = travelOptions;
         this.roundTripTimeMillis = roundTripTimeMillis;
         this.parseTimeMillis 	 = parseTimeMillis;
     }
 
-    public void setCode(String code) {
```

Comment:
If this class is abstract, wouldn't be better to keep the method modifiers protected at least?
Sorry for the question, but I didn't understand the change to default modifier.

Reply:
package-private (default) is stricter than protected (see https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html)

If it turns out later that we need access to that method from the sub classes we can change it to protected